src/core/emergency-action.c

```diff
@@ -27,11 +39,23 @@ void emergency_action(
                 const char *reboot_arg,
                 int exit_status,
                 const char *reason) {
+        Unit *u;
 
         assert(m);
         assert(action >= 0);
         assert(action < _EMERGENCY_ACTION_MAX);
 
+        /* Is the special shutdown target active or queued? If so, we are in shutdown state */
+        u = manager_get_unit(m, SPECIAL_SHUTDOWN_TARGET);
+        if ((u && unit_active_or_pending(u))) {
+                if ( (action == EMERGENCY_ACTION_REBOOT) ||
+                     (action == EMERGENCY_ACTION_POWEROFF) ) {
+                        log_info("EmergencyAction Special Shutdown Tgt is active Skip %s",
```

Comment:
Please no abbreviations like this. Use log_unit_info with a normal sentence.

Reply:
It does not look like emergency_action is built to use log_unit_info. If I remove the call to manager_get_unit() to use manager_state() there would be no unit available to call log_unit_info(). I can do as you request but I think I would still have to call manager_get_unit() in order to use log_unit_info(). Is that the way you would like to go with this?